clang/lib/Sema/SemaExpr.cpp

```diff
@@ -13389,6 +13389,25 @@ QualType Sema::CheckVectorLogicalOperands(ExprResult &LHS, ExprResult &RHS,
   return GetSignedVectorType(LHS.get()->getType());
 }
 
+QualType Sema::CheckMatrixLogicalOperands(ExprResult &LHS, ExprResult &RHS,
+                                          SourceLocation Loc,
+                                          BinaryOperatorKind Opc) {
+
+  if (!getLangOpts().HLSL) {
+    assert(false && "Logical operands are not support in C\\C++");
+    return QualType();
+  }
```

Comment:
This case is an assert where as the 2018 HLSL case is a Sema diagnostic error because its impossible today for C\C++ code to get into this function without explicit support for logical operators. The HLSL code really only exists to  disable logical operators for the 2021 language mode. The 2018 language mode is suppose to support them but thats not our default language mode right now so that work will be punted on for now.